src/Google/Ads/GoogleAds/Util/V1/GoogleAdsFailures.php

```diff
@@ -36,8 +34,10 @@ public static function fromAny(Any $any)
     {
         // This initialization is needed to populate the descriptor pool with the GoogleAdsFailure
         // class and prevent exceptions from being thrown.
-        if (!isset(self::$unusedGoogleAdsFailure)) {
-            self::$unusedGoogleAdsFailure = new GoogleAdsFailure();
+        if (is_null(
+            DescriptorPool::getGeneratedPool()->getDescriptorByClassName(GoogleAdsFailure::class)
```

Comment:
1) Have you confirmed with the Protobuf team that this is a reliable way to fix?
2) The generated pool will always retain all descriptors in a run time until the program finishes?

Reply:
1. I have not confirmed directly with Protobuf team, this is based on a discussion I got with Bob.
2. I do not have the answer to this question, actually the purpose of this edit is to make sure we do not have to know: I basically check and instantiate a new object anytime I cannot find the descriptor that is necessary in the pool that is being used (singleton) by Any to unpack. This is board-effect free as it works regardless of how PHP, Protobuf or any other dependencies behave at runtime. The current test assumes that instantiating once the object will be enough to load the descriptor pool forever, this may be true for most cases today but there might be corner cases we do not know and/or it could change in future for unforeseeable reasons.

Reply:
Point no. 2 here seems pretty solid to me, it really looks better to set the descriptor in the pool directly rather than relying on variable instantiation doing so implicitly.

Reply:
SGTM.